### PR TITLE
Draft 1: An example of reading all the data needed in a test from a single fixture

### PR DESCRIPTION
**Pull request by TheBestPessimist**
https://github.com/metasfresh/metasfresh-e2e/issues/286

### CODE REVIEW

File: cypress/support/utils/pricelist.js [hunk 1]
```diff
@@ -80,7 +80,6 @@ export class PriceList {
     describe(`Create new PriceListVersion ${priceListVersion.name}`, function() {
       cy.selectTab('M_PriceList_Version');
       cy.pressAddNewButton();
-      cy.writeIntoStringField('Name', `${priceListVersion.name} ${priceListVersion.validFrom}`, true, null, true);
       cy.writeIntoStringField('ValidFrom', priceListVersion.validFrom, true, null, true);
```
Comment (cristinastefan90) [inline cypress/support/utils/pricelist.js]: I have just tried to add a name and it works, it isn't overriden: 
image. Let's have a chat on Monday.

File: cypress/integration/aaaaaaaaaaaaaaaaaaaaaaaa/aaaaaaaaaaaa.js [hunk 2]
```diff
@@ -43,41 +51,57 @@ describe('Read fixture and prepare test data', function() {
...
-  //
-  //   Builder2.createProductEntities(productCategoryJson, priceListName, productName1, productName1, productType);
-  // });
+  it('Create Product1 and Category', function() {
+    const productCategoryJson = fixture['ProductCategory'];
+    productCategoryJson.name = productCategoryJson.name + appendDate();
+    categoryName = productCategoryJson.name;
+
+    const productJson = fixture['Products'][0];
+    productJson.name = productJson.name + appendDate();
```
Comment (cristinastefan90) [inline cypress/integration/aaaaaaaaaaaaaaaaaaaaaaaa/aaaaaaaaaaaa.js]: Can't we put appendDate() directly int he fixture?

File: cypress/support/utils/pricelist.js (hunk 1 above)
Comment (TheBestPessimist) [reply cypress/support/utils/pricelist.js]: The issue is that in Product Prices, or when filtering by a PLV the "autogenerated" name is used, therefore what is set manually as PLV name is useless imo. eg.: in the picture i could have named my PLV "PricelistFor<test name here> 1", and also have a second one  "PricelistFor<test name here> 2", but that would not be seen here in the selection dialog:
image

File: cypress/integration/aaaaaaaaaaaaaaaaaaaaaaaa/aaaaaaaaaaaa.js (hunk 2 above)
Comment (TheBestPessimist) [reply cypress/integration/aaaaaaaaaaaaaaaaaaaaaaaa/aaaaaaaaaaaa.js]: I don't think we can. If we add a specific date, then all future runs have the same date, so our test data is no longer unique.

If what you ask is to make a name in the fixture like "Product1_`+ appendDate()`" then i guess we could do that but using [eval](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval) is a can of worms i will not touch even wearing 2 hazard suits at the same time ☣️ and will not approve of code using that.

**Comment by TheBestPessimist** [comment]
Closing as it is too difficult to use

**Pull request closed without merging**